kartothek/io/dask/_update.py

```diff
@@ -35,6 +37,40 @@ def _hash_bucket(df: pd.DataFrame, subset: List[str], num_buckets: int):
     return df.assign(**{_KTK_HASH_BUCKET: buckets.astype(f"uint{bit_width}")})
 
 
+def _pack_payload(partition: pd.DataFrame, group_key: List[str]) -> pd.DataFrame:
+    # TODO: benchmark how good plain parquet would do
+    try:
+        # Technically distributed is an optional dependency
+        from distributed.protocol import serialize_bytes
+    except ImportError:
+        _logger.warning(
+            "Shuffle payload columns cannot be compressed since distributed is not installed."
+        )
+        return partition
+
+    # This is not _optimal_ since dask internally will perform another groupby in shuffle-group
+    # Once / if this is merged upstream, we can get rid of this
+    res = partition.groupby(group_key, sort=False, observed=True).apply(serialize_bytes)
+    res.name = _PAYLOAD_COL
+    return res.reset_index()
+
+
+def _unpack_payload(partition: pd.DataFrame) -> pd.DataFrame:
+    try:
+        # Technically distributed is an optional dependency
+        from distributed.protocol import deserialize_bytes
+    except ImportError:
+        _logger.warning(
+            "Shuffle payload columns cannot be compressed since distributed is not installed."
+        )
+        return partition
+
+    return pd.concat(
+        partition[_PAYLOAD_COL].map(lambda part: deserialize_bytes(part)).values,
```

Comment:
style np
```suggestion
        partition[_PAYLOAD_COL].map(deserialize_bytes).values,
```

Reply:
oops. slipped in from an earlier iteration